src/transformers/modeling_utils.py

```diff
@@ -1876,7 +1876,18 @@ def gradient_checkpointing_enable(self, gradient_checkpointing_kwargs=None):
 
         gradient_checkpointing_func = functools.partial(checkpoint, **gradient_checkpointing_kwargs)
 
-        self._set_gradient_checkpointing(enable=True, gradient_checkpointing_func=gradient_checkpointing_func)
+        # For old GC format (transformers < 4.35.0) for models that live on the Hub
+        # we will fall back to the overwritten `_set_gradient_checkpointing` methid
+        _is_using_old_format = "value" in inspect.signature(self._set_gradient_checkpointing).parameters
+
+        if not _is_using_old_format:
+            self._set_gradient_checkpointing(enable=True, gradient_checkpointing_func=gradient_checkpointing_func)
+        else:
+            self.apply(partial(self._set_gradient_checkpointing, value=True))
+            logger.warn(
+                "You are using an old version of the checkpointing format that is deprecated (We will also silently ignore `gradient_checkpointing_kwargs` in case you passed it)."
+                "Please update to the new format on your modeling file. To use the new format, you need to completely remove the definition of the method `_set_gradient_checkpointing` in your model."
+            )
```

Comment:
we should raise an error IMO, saying that an issue should be opened on the hub for this particular model!

Reply:
Raising an error would make GC totally unusable, so maybe too brutal, WDYT?

Reply:
Agreed, but it's more visible for people to open issues on the repo and maintainers to update it